File: front/src/enums/EStatusCommande.ts
export enum EStatusCommande
{
    Valider,
    EnAttenteValidation,
    Annuler,
    Livrer,
    Tout
}

export class ConvertionEnum
{
    public static StatusCommande(_status: EStatusCommande): string
    {
        switch (_status) 
        {
            case EStatusCommande.Valider:
                return "Valider"

            case EStatusCommande.Annuler:
                return "Annuler"

            case EStatusCommande.Livrer:
                return "Livrer"

            case EStatusCommande.EnAttenteValidation:
                return "A valider"

            case EStatusCommande.Tout:
                return "Tout"
        }
    }
}

File: Api/src/Api/Enums/EStatusCommande.cs
using System.ComponentModel;

namespace Api.Enums;

public enum EStatusCommande
{
    Valider,

    [Description("En attente de validation")]
    EnAttenteValidation,
    Annuler,
    Livrer,
    Tout
}

public enum EStatusCommandeModifier
{
    Valider = 0,
    Annuler = 2,
    Livrer = 3
}
